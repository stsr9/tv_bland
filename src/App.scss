:root {
  --color-primary: #333;
  --breakpoint-large: 1600px;
  --font-size-small: .8em
}

.app {
  text-align: left;
}

h1, h2 {
  margin: 0;
}

.logo {
  cursor: pointer;
}

.app-header {
  padding: 5vh 10vw 0 10vw;
  background-color: lighten(lightgray, 10%);
  @media (max-width: 800px) {
    padding: 5vh 5vw 0 5vw;
  }
}

.view-header {
  background-color: lighten(lightgray, 10%);
  padding: 5vh 10vw 15vh 10vw;
  @media (max-width: 800px) {
    padding: 5vh 5vw 0 5vw;
  }
}

.view-content {
  padding: 0 10vw 5vh 10vw;
  @media (max-width: 800px) {
    padding: 0 5vw 5vh 5vw;
  }
}

.flex {
  display: flex;
}

.flex-column {
  display: flex;
  flex-direction: column;
}

.fwn {
  font-weight: normal;
}

.fwb {
  font-weight: bold;
}

.dtc {
  color: var(--color-primary)
}

