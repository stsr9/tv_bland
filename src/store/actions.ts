import { Schedule } from '../models/schedule'
import ScheduleAPI from '../api/schedule-api'
import ShowAPI from '../api/shows-api'
import { Show } from '../models/show'
import { Actor } from '../models/actor'

export enum ActionTypes {
    fetchEpisodes = 'FETCH_EPISODES',
    fetchEpisodeDetails = 'FETCH_EPISODE_DETAILS',
	fetchCast = 'FETCH_CAST',
	clearDetails = 'CLEAR_DETAILS'
}

interface EpisodeActions {
	fetchEpisodes: Function
	fetchEpisodeDetails: Function
	clearDetails: Function
}

interface CastActions {
	fetchCast: Function
}

export const episodeActions: EpisodeActions = {
	fetchEpisodes: () => async(dispatch: Function) => {
		const episodes: Schedule[] = await ScheduleAPI.fetchAllEpisodes()
	    dispatch({
			type: ActionTypes.fetchEpisodes,
			episodes
	    })
	},
	fetchEpisodeDetails: (id: string) => async(dispatch: Function) => {
		const episodeDetails: Show = await ShowAPI.fetchEpisodeDetails(id)
	    dispatch({
			type: ActionTypes.fetchEpisodeDetails,
			episodeDetails
	    })
	},
	clearDetails: () => (dispatch: Function) => {
		dispatch({
			type: ActionTypes.clearDetails,
	    })
	}
}

export const castActions: CastActions = {
	fetchCast: (id: string) => async(dispatch: Function) => {
		const cast: Actor[] = await ShowAPI.showCast(id)
	    dispatch({
			type: ActionTypes.fetchCast,
			cast
	    })
	}
}
