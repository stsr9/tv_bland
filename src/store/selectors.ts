import { MainState } from 'models/main-state'

export const episodeSelector = {
	getEpisodeList: ({ episodesState }: MainState) => episodesState.episodes,
	getEpisodeDetails: ({ episodesState }: MainState) =>
		episodesState.episodeDetails
}

export const castSelector = {
	getCast: ({ castState }: MainState) => castState.cast
}
