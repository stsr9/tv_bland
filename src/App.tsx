import './App.scss'
import { BrowserRouter as Router } from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import 'font-awesome/css/font-awesome.min.css';

import {
	ROUTES,
	RenderRoutes,
} from './routes/routes'

function App(props: any) {
    const { t } = useTranslation('common')
    return (
        <div className='app'>
            <header className='app-header'>
                <h2>{t('mainTitle')}</h2>
            </header>
            <Router>
                <RenderRoutes {...props} routes={ROUTES} />
            </Router>
        </div>
    )
}

export default App
