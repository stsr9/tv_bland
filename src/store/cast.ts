import { AnyAction } from 'redux'
import { ActionTypes } from './actions'
import { Actor } from 'models/actor'

export interface InitialState {
	cast: Actor[]
}

const initialState: InitialState = {
	cast: []
}

export const castState = (
	state: InitialState = initialState,
	action: AnyAction
) => {
	switch (action.type) {
		case ActionTypes.fetchCast:
			return {
				...state,
				cast: action.cast
			}
		default:
			return state
	}
}
