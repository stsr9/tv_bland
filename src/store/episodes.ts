import { AnyAction } from 'redux';
import { Schedule } from '../models/schedule'
import { Show } from '../models/show';
import { ActionTypes } from './actions'

type InitialState = {
	episodes: Schedule[]
	episodeDetails: Show | null
}

const initialState: InitialState = {
	episodes: [],
	episodeDetails: null
};

export const episodesState =  (state: InitialState = initialState, action: AnyAction) => {
	switch (action.type) {
		case ActionTypes.fetchEpisodes:
			return {
				...state,
                episodes: action.episodes
			};
		case ActionTypes.fetchEpisodeDetails:
			return {
				...state,
                episodeDetails: action.episodeDetails
			};
		case ActionTypes.clearDetails:
			return {
				...state,
                episodeDetails: null
			};
		default:
			return state;
	}
};
