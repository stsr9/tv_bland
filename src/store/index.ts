import { combineReducers } from 'redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { episodesState } from './episodes'
import { castState } from './cast'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const appReducers = combineReducers({
	episodesState,
	castState
});

export const store = createStore(
	appReducers,
	composeEnhancers(applyMiddleware(thunk))
);

