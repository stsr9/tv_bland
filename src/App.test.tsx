import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore } from 'redux'
import { appReducers } from './store'
import App from './App'
import thunk from 'redux-thunk'
import { I18nextProvider } from 'react-i18next'
import i18next from 'i18next'
import common_en from './translations/en/common.json'

const store = createStore(appReducers, applyMiddleware(thunk))

i18next.init({
	interpolation: { escapeValue: false },
	lng: 'en',
	resources: {
		en: {
			common: common_en
		}
	}
})

it('renders without crashing', () => {
	const app = renderer
		.create(
			<I18nextProvider i18n={i18next}>
				<Provider store={store}>
					<App />
				</Provider>
			</I18nextProvider>
		)
		.toJSON()

	expect(app).toMatchSnapshot()
})
