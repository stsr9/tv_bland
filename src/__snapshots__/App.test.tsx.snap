// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app"
>
  <header
    className="app-header"
  >
    <h2
      className="logo"
      onClick={[Function]}
    >
      TV Bland
    </h2>
  </header>
  <div
    className="splash-screen"
  >
    <h1>
      TV BLAND
    </h1>
  </div>
</div>
`;
